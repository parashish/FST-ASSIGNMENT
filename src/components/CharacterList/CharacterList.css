.card {
  box-sizing: border-box;
  border: 5px solid black;
  position: relative;
  width: 50%;
  background: #7f7f7f;
}
.card img {
  width: 100%;
}

.card .transparent {
  background-color: rgba(10, 0, 0, 0.5);
  margin-top: -60px;
  position: absolute;
  width: 100%;
  height: 60px;
  color: #ccc;
}

.transparent p {
  margin: 0;
  padding: 2px;
}

.character-info {
  list-style-type: none;
  padding: 0;
  margin: 0;
  background-color: grey;
}

.character-info li {
  padding: 16px 8px 16px 8px;
  display: flex;
  align-content: flex-end;
  justify-content: space-between;
  border-bottom: 1px solid #6e6a6a;
  color: antiquewhite;
}

.character-info li:last-child {
  border-bottom: 0;
}
.character-info li span {
  color: #ff9a00;
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: end;
}

@media (min-width: 768px) and (max-width: 1024px) {
  .card {
    border: 5px solid black;
    position: relative;
    width: 25%;
    background: #7f7f7f;
  }
}

@media only screen and (min-width: 1024px) {
  .card {
    border: 5px solid black;
    position: relative;
    width: 25%;
    background: #7f7f7f;
  }
}
