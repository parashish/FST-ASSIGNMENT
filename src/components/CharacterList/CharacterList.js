import React from 'react';
import './CharacterList.css';

const CharacterList = (props) => {
  const {
    image,
    name,
    status,
    species,
    gender,
    origin,
    location,
    created,
  } = props.character;

  const getIdCreationYears = () => {
    return `Created ${
      parseInt(new Date().getFullYear()) -
      parseInt(new Date(created).getFullYear())
    } years ago`;
  };

  return (
    <React.Fragment>
      <div id="id" className="card">
        <img src={image} alt={name} />
        <div className="transparent">
          <p>{name}</p>
          <p>ID {getIdCreationYears()}</p>
        </div>
        <ul className="character-info">
          <li>
            Status: <span>{status} </span>
          </li>
          <li>
            Species: <span> {species}</span>
          </li>
          <li>
            Gender: <span>{gender}</span>
          </li>
          <li>
            Origin: <span>{origin.name}</span>
          </li>
          <li>
            Last Location:<span>{location.name}</span>
          </li>
        </ul>
      </div>
    </React.Fragment>
  );
};

export default CharacterList;
