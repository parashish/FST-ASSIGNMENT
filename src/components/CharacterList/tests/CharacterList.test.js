import React from 'react';
import { shallow } from 'enzyme';
import CharacterList from '../CharacterList';

describe('CharacterList component', () => {
  const props = {
    character: {
      created: '2017-11-04T18:48:46.250Z',
      episode: [
        'https://rickandmortyapi.com/api/episode/1',
        'https://rickandmortyapi.com/api/episode/2',
        'https://rickandmortyapi.com/api/episode/3',
      ],
      gender: 'Male',
      id: 1,
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      location: {
        name: 'Earth (Replacement Dimension)',
        url: 'https://rickandmortyapi.com/api/location/20',
      },
      name: 'Rick Sanchez',
      origin: {
        name: 'Earth (C-137)',
        url: 'https://rickandmortyapi.com/api/location/1',
      },
      species: 'Human',
      status: 'Alive',
      type: '',
      url: 'https://rickandmortyapi.com/api/character/1',
    },
  };
  it('should render correctly with props` and match snapshot', () => {
    const component = shallow(<CharacterList {...props} />);
    expect(component).toMatchSnapshot();
  });
});
