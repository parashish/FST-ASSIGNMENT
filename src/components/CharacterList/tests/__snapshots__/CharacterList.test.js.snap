// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component should render correctly with props\` and match snapshot 1`] = `
<Fragment>
  <div
    className="card"
  >
    <img
      alt="Rick Sanchez"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
    <div
      className="transparent"
    >
      <p>
        Rick Sanchez
      </p>
      <p>
        ID:
        Created 3 years ago
      </p>
    </div>
    <ul
      className="character-info"
    >
      <li>
        Status:
        <span>
          Alive
        </span>
      </li>
      <li>
        Species:
        <span>
          Human
        </span>
      </li>
      <li>
        Gender:
        <span>
          Male
        </span>
      </li>
      <li>
        Origin:
        <span>
          Earth (C-137)
        </span>
      </li>
      <li>
        Last Location:
        <span>
          Earth (Replacement Dimension)
        </span>
      </li>
    </ul>
  </div>
</Fragment>
`;
